### PR TITLE
feat: add Docker & Docker Compose For Dev Mode.

### PR DESCRIPTION
**Pull request by geekyharsh05**
This pr resolves the issue: #4

This pr setups:
-  Docker compose
-  Integrates postgres db in the compose
-  Example .env file for docker environment variables.
-  Multi-Stage Build for less image size.
-  Few Changes in the scripts.

This is the local docker-compose results, everything is working fine. <br />

Screenshot 2024-10-01 at 10 37 25 PM

### Sign-Up User
Screenshot 2024-10-01 at 10 03 17 PM

### Sign-In User
Screenshot 2024-10-01 at 10 04 13 PM

### Get Logged In User (/me)
Screenshot 2024-10-01 at 10 18 41 PM

I have tested it on my end. Verified that services start correctly using `docker-compose up`.

@jaspermayone Please review the Docker Compose configuration and let me know if any adjustments are needed.

### CODE REVIEW

**Comment by jaspermayone** [comment]
Will take a look now!

**Comment by geekyharsh05** [comment]
@jaspermayone Any updates?

**Comment by jaspermayone** [comment]
I've been busy, will review in the next 48 hours

**Comment by geekyharsh05** [comment]
@jaspermayone okay no problem!

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   ],
   "packageManager": "pnpm@9.5.0",
   "scripts": {
+    "preinstall": "npx only-allow pnpm",
```
Comment (jaspermayone) [inline package.json:15]: Why do we need this @geekyharsh05?

**Review by jaspermayone — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (geekyharsh05) [reply package.json:15]: To allow only pnpm as the package manager. If accidentally someone runs yarn install, it'll prompt a msg and say use pnpm install.

**Review by jaspermayone — APPROVED** [review]

**Pull request merged**